backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,33 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function putTransaction($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        $cardList = Listing::find($card_id);
+        if ($card->isUserOwner($user)) {
+            return response()->build(self::RESPONSE_MESSAGE_ERROR_UNAUTHORIZED, 'User does not own the card');
+        }
+
+        $data = json_decode(Request::getContent(), true);
+        $card->user_id = $user->id;
+
+        $card->save();
+
+        $transaction = new Transaction();
+        $transaction->card_id = $card_id;
+        $transaction->user_id = $user->id;
+        $transaction->price = 0;
```

Comment:
This card listing already exists in the database and holds the price for the listing:

`$table->bigInteger('price')->unsigned();`

https://github.com/CryptoCardsETH/cryptocards/blob/master/backend/database/migrations/2018_02_18_015214_listings.php#L22